ParquetHadoop/build.gradle

```diff
@@ -36,4 +35,8 @@ dependencies {
      *    [...]
      */
     runtimeOnly('com.fasterxml.woodstox:woodstox-core:5.2.1')
+    runtimeOnly('org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1')
+    runtime('com.hadoop.gplcompression:hadoop-lzo:0.4.20')
+    runtimeOnly('org.lz4:lz4-java:1.8.0')
+    runtimeOnly('com.github.rdblue:brotli-codec:0.1.1')
```

Comment:
Are these dependencies that hadoop-common depends on? I'm worried that the transitive = false is biting us now.

Are these dependencies necessary; or optional, depending on how the data is encoded?

Reply:
When we write a table in parquet format, they are pulled in dynamically (runtime dependency) depending on the compression codec specified for writing.  Traditionally we had supported only SNAPPY has a codec, but this diff is adding support to switch the default codec and LZ4, LZO, GZIP, BROTLI and ZSTD.
A later diff will make the compression codec part of an optional argument part of type ParquetWriteInstructions, that can give general instructions for writing (including other things like eg, remap column names).